### DISCUSSION TITLE
How to add CoAP .well-known/core endpoint?

### DISCUSSION THREAD

**Discussion by YSLew-tado** [openthread/openthread#9023] [Q&A]
Hi,

I would like to add a CoAP _.well-known/core_ endpoint to my device. However, I have not found any API to fetch the registered resources.

Am I missing the right API?

Thanks.
 
Sources: https://github.com/openthread/openthread/blob/main/src/core/coap/coap.cpp, https://github.com/openthread/openthread/blob/main/src/core/coap/coap.hpp

**Comment by jwhui** [comment]
OpenThread does not implement CoRE Resource Discovery or Link Format.

**Comment by rretanubun** [comment]
@YSLew-tado : Have a look at https://github.com/obgm/libcoap (which I think auto generates the contents of `./well-known/core` as you register additional coap endpoints handlers to it. You can integrate this into  your `openThread codebase` and uses it for your needs.
